frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1367,11 +1384,25 @@ const IR::Node* TypeInference::postorder(IR::Type_Header* type) {
                           varbit, field);
                 return type;
             }
+        } else if (ftype->is<IR::Type_Struct>()) {
```

Comment:
This whole block is probably no needed. validateFields() call on line 1372 has already checked that, as the validator passed to it only allows bit-vector structs.

Reply:
That's correct.  I will remove the code - thanks.